request.go

```diff
@@ -508,3 +520,212 @@ func encodeSQLBind(enc *msgpack.Encoder, from interface{}) error {
 	}
 	return nil
 }
+
+// Request is an interface that provides the necessary data to create a request
+// that will be sent to a tarantool instance.
+type Request interface {
+	// Code returns a IPROTO code for the request.
+	Code() int32
+	// BodyFunc returns a functions that can fill an encoder with
+	// the request body or it returns an error if unable to create the function.
+	BodyFunc(resolver SchemaResolver) (func(enc *msgpack.Encoder) error, error)
+}
+
+type baseRequest struct {
+	requestCode int32
+}
+
+// Code returns a IPROTO code for the request.
+func (req *baseRequest) Code() int32 {
+	return req.requestCode
+}
+
+type spaceRequest struct {
+	baseRequest
+	space interface{}
+}
+
+func (req *spaceRequest) setSpace(space interface{}) {
+	req.space = space
+}
+
+type spaceIndexRequest struct {
+	spaceRequest
+	index interface{}
+}
+
+func (req *spaceIndexRequest) setIndex(index interface{}) {
+	req.index = index
+}
+
+// SelectRequest allows you to create a select request object for execution
+// by a Connection.
+type SelectRequest struct {
+	spaceIndexRequest
+	offset, limit, iterator uint32
+	key                     interface{}
+}
+
+// NewSelectRequest returns a new empty SelectRequest.
+func NewSelectRequest(space interface{}) *SelectRequest {
+	req := new(SelectRequest)
+	req.requestCode = SelectRequestCode
+	req.setSpace(space)
+	req.key = []interface{}{}
+	return req
+}
+
+// Index sets the index for the select request.
+// Note: default value is 0.
+func (req *SelectRequest) Index(index interface{}) *SelectRequest {
+	req.setIndex(index)
+	return req
+}
+
+// Offset sets the offset for the select request.
+// Note: default value is 0.
+func (req *SelectRequest) Offset(offset uint32) *SelectRequest {
+	req.offset = offset
+	return req
+}
+
+// Limit sets the limit for the select request.
+// Note: default value is 0.
+func (req *SelectRequest) Limit(limit uint32) *SelectRequest {
+	req.limit = limit
+	return req
+}
+
+// Iterator set the iterator for the select request.
+// Note: default value is IterEq.
```

Comment:
Should not it be `IterEQ` when the key is provided and is not empty and `IterAll` otherwise?

I just seen the [similar logic](https://github.com/tarantool/tarantool/blob/ede831d317c3a5ba02f202a7daf387742ddbfeb6/src/box/lua/schema.lua#L1892-L1893) in tarantool's Lua APi implementation and wonder, whether something like this should be there.

Reply:
Good idea, done.

Reply:
What if the key is set to `null` or an empty array? I don't know, to be honest, whether it makes any difference in responses. If not, feel free to choose any solution.

Reply:
It seems to me that we should not try to validate the input parameters here. As we have not tried to do this anywhere before (`Connection.Select` as example).

The user can still shoot himself in the foot no matter how hard we try. It will be clear not to validate the input parameters for all cases.

Reply:
Okay.